web/pdf_viewer.js

```diff
@@ -1099,10 +1112,22 @@ class PDFViewer {
       newScale * PixelsPerInch.PDF_TO_CSS_UNITS
     );
 
-    const updateArgs = { scale: newScale };
-    for (const pageView of this._pages) {
-      pageView.update(updateArgs);
+    const mustPostponeDrawing = drawingDelay >= 0 && drawingDelay < 1000;
+    const updateArgs = {
+      scale: newScale,
+    };
+    if (mustPostponeDrawing) {
+      updateArgs.drawingDelay = drawingDelay;
+    }
+    this.refresh(false, updateArgs);
+
+    if (mustPostponeDrawing) {
```

Comment:
With the delayed rendering we're no longer toggling the `loadingIcon`-div on the pages until the actual re-rendering starts. It feels a little surprising to re-size the page, using the CSS-only zooming, and then *after* the `drawingDelay` having the indicator suddenly show up.

Don't we still want to keep showing the `loadingIcon`-div, like in the current code, immediately when rendering starts?
If so, we could probably fix that here by adding something like 
```suggestion
    if (mustPostponeDrawing) {
      const visiblePages = this._getVisiblePages();
      this.#toggleLoadingIconSpinner(visiblePages.ids);
      
```

Reply:
In general, I'd be rather in favor of just displaying the loading icon only when the page redrawing is taking more than 1 second (or something like that) just to inform the user that the viewer is not stuck. 
I don't really see how this spinner could be useful for the user especially when they're zooming.